ra/ra_test.go

```diff
@@ -1669,56 +1669,65 @@ func TestNewOrder_AuthzReuse_NoPending(t *testing.T) {
-		"test": NewValidationProfile(nil),
-	}
-
-	// Issuance should succeed.
-	orderReq := &rapb.NewOrderRequest{
-		RegistrationID:         Registration.Id,
-		DnsNames:               []string{"a.example.com"},
-		CertificateProfileName: "test",
+	testCases := []struct {
+		name              string
+		allowList         *allowlist.List[int64]
+		expectErr         bool
+		expectErrContains string
+	}{
+		{
+			name:      "Allow All Account IDs",
+			allowList: nil,
+			expectErr: false,
+		},
+		{
+			name:              "Deny All But Account ID 1337",
+			allowList:         allowlist.NewList([]int64{1337}),
+			expectErr:         true,
+			expectErrContains: "not permitted to use certificate profile",
+		},
+		{
+			name:              "Deny All",
+			allowList:         allowlist.NewList([]int64{}),
+			expectErr:         true,
+			expectErrContains: "not permitted to use certificate profile",
+		},
+		{
+			name:      "Allow Registration ID",
+			allowList: allowlist.NewList([]int64{Registration.Id}),
+			expectErr: false,
+		},
 	}
-	_, err := ra.NewOrder(context.Background(), orderReq)
-	test.AssertNotError(t, err, "NewOrder for account ID that is on the allowlist failed")
 
-	// Set up a ValidationProfile with an allowList that doesn't include
-	// Registration.Id.
-	ra.validationProfiles = map[string]*ValidationProfile{
-		"test": NewValidationProfile(allowlist.NewList([]int64{1337})),
-	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
 
-	// Issuance should fail with an unauthorized error regarding the certificate
-	// profile.
-	orderReq = &rapb.NewOrderRequest{
-		RegistrationID:         Registration.Id,
-		DnsNames:               []string{"b.example.com"},
-		CertificateProfileName: "test",
-	}
-	_, err = ra.NewOrder(context.Background(), orderReq)
-	test.AssertError(t, err, "NewOrder with invalid profile did not error")
-	test.AssertErrorIs(t, err, berrors.Unauthorized)
-	test.AssertContains(t, err.Error(), "not permitted to use certificate profile")
+			ra.validationProfiles = map[string]*ValidationProfile{
+				"test": NewValidationProfile(tc.allowList),
+			}
 
-	// Set up a ValidationProfile with an allowList that does include
-	// Registration.Id.
-	ra.validationProfiles = map[string]*ValidationProfile{
-		"test": NewValidationProfile(allowlist.NewList([]int64{Registration.Id})),
-	}
+			orderReq := &rapb.NewOrderRequest{
+				RegistrationID:         Registration.Id,
+				DnsNames:               []string{randomDomain()},
+				CertificateProfileName: "test",
+			}
+			_, err := ra.NewOrder(context.Background(), orderReq)
 
-	// Issuance should succeed.
-	orderReq = &rapb.NewOrderRequest{
-		RegistrationID:         Registration.Id,
-		DnsNames:               []string{"c.example.com"},
-		CertificateProfileName: "test",
+			if tc.expectErr {
+				test.AssertError(t, err, "NewOrder did not error")
+				test.AssertErrorIs(t, err, berrors.Unauthorized)
```

Comment:
The `expectErr` boolean is redundant with the `expectErrContains` string, and the AssertError call is redundant with the AssertErrorIs (which will fail if the error is nil):
```suggestion
			if tc.expectErrContains != "" {
				test.AssertErrorIs(t, err, berrors.Unauthorized)
```